src/Tests/dotnet-workload-install.Tests/GivenNetSdkManagedWorkloadInstall.cs

```diff
@@ -0,0 +1,255 @@
+        }
+
+        [Fact]
+        public void GivenManagedInstallItCanRollBackInstallFailures()
+        {
+            var version = "6.0.100";
+            var (dotnetRoot, installer, nugetInstaller) = GetTestInstaller(failingInstaller: true);
+            var packInfo = new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk, Path.Combine(dotnetRoot, "packs", "Xamarin.Android.Sdk", "8.4.7"));
+            try
+            {
+                installer.InstallWorkloadPack(packInfo, new SdkFeatureBand(version));
+
+                // Install should have failed
+                true.Should().BeFalse();
+            }
+            catch (Exception e)
+            {
+                var failingNugetInstaller = nugetInstaller as FailingNuGetPackageInstaller;
+
+                e.Message.Should().Be("Test Failure");
+                // Nupkgs should be removed
+                Directory.Exists(failingNugetInstaller.MockPackageDir).Should().BeFalse();
+                // Packs should be removed
+                Directory.Exists(packInfo.Path).Should().BeFalse();
+            }
+        }
+
+        [Fact]
+        public void GivenManagedInstallItCanGarbageCollect()
+        {
+            var (dotnetRoot, installer, _) = GetTestInstaller();
+            var packs = new PackInfo[]
+            {
+                new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Library, Path.Combine(dotnetRoot, "packs", "Xamarin.Android.Sdk", "8.4.7")),
+                new PackInfo("Xamarin.Android.Framework", "8.4", WorkloadPackKind.Framework, Path.Combine(dotnetRoot, "packs", "Xamarin.Android.Framework", "8.4"))
+            };
+            var sdkVersions = new string[] { "6.0.100", "6.0.300" };
+
+            // Write fake packs
+            var installedPacksPath = Path.Combine(dotnetRoot, "metadata", "workloads", "InstalledPacks");
+            foreach (var sdkVersion in sdkVersions)
+            {
+                Directory.CreateDirectory(Path.Combine(dotnetRoot, "metadata", "workloads", sdkVersion, "InstalledWorkloads"));
+                foreach (var pack in packs)
+                {
+                    var packRecordPath = Path.Combine(installedPacksPath, pack.Id, pack.Version, sdkVersion);
+                    Directory.CreateDirectory(Path.GetDirectoryName(packRecordPath));
+                    File.WriteAllText(packRecordPath, string.Empty);
+                    Directory.CreateDirectory(pack.Path);
+                }
+            }
+
+            installer.GarbageCollectInstalledWorkloadPacks();
+
+            Directory.EnumerateFileSystemEntries(installedPacksPath)
+                .Should()
+                .BeEmpty();
+            foreach (var pack in packs)
+            {
+                Directory.Exists(pack.Path)
+                    .Should()
+                    .BeFalse();
+            }
+        }
+
+        [Fact]
+        public void GivenManagedInstallItDoesNotRemovePacksWithInstallRecords()
```

Comment:
Can we add some packs to this test that aren't part of the `xamarin-android-build` workload and verify that they get garbage collected?